src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,69 @@
+package duke.task;
+
+import duke.exception.DukeException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+
+/**
+ * This class inherits from the abstract Task class
+ * and encapsulates the logic of a Deadline task.
+ */
+public class Deadline extends Task {
+    /* Due Date field */
+    private LocalDateTime dueDate;
+
+    /**
+     * Constructor for the Deadline Task.
+     *
+     * @param description description of the task.
+     */
+    public Deadline(String description, LocalDateTime dueDate) {
+        super(description);
+        this.dueDate = dueDate;
+    }
+
+    /**
+     * Factory Method for the construction of a Deadline for user input.
+     *
+     * @param in user's input.
+     * @return Deadline with the relevant input fields.
+     * @throws DukeException if no task or incorrect formatting is given.
+     */
+    public static Deadline createDeadline(String in) throws DukeException {
+        String[] temp = in.split(" */by* ");
```

Comment:
You might want to rename `temp` to provide more information about the content of the variable. Do consider fixing similar issues in other classes as well.